integration_tests/contracts/contracts/TestMaliciousSupply.sol

```diff
@@ -0,0 +1,27 @@
+pragma solidity 0.8.10;
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract TestMaliciousSupply is ERC20 {
+	uint256 public count;
+
+	constructor() public ERC20("MAX", "MAX") {
+		_mint(msg.sender, 115792089237316195423570985008687907853269984665640564039457584007913129639935);
+	}
+
+	function transferFrom(
+		address from,
+		address to,
+		uint256 amount
+	) public virtual override returns (bool) {
+		count = count + 1;
+		return true;
+	}
+
+	function balanceOf(address account) public view virtual override returns (uint256) {
+		if (count % 2 == 0) {
+			return 0;
+		} else {
+			return 115792089237316195423570985008687907853269984665640564039457584007913129639935;
+		}
```

Comment:
not sure about the alternatives.

because the gravity bridge records in the event its difference of balance (balanceOf is called before the transfer, and balanceOf is called after the transfer, the event record the difference) we need to find a trick that allows us to get a diff equals to max int.

This trick works but it may be a bit hard to understand. I can add some comment in the test case